src/Refactoring-UI/RBRenameVariableDriver.class.st

```diff
@@ -0,0 +1,134 @@
+"
+I'm an abstract model for user interaction for the rename variables refactoring.
+"
+Class {
+	#name : 'RBRenameVariableDriver',
+	#superclass : 'RBInteractionDriver',
+	#instVars : [
+		'oldName',
+		'newName',
+		'lastName',
+		'class',
+		'shouldEscape'
+	],
+	#category : 'Refactoring-UI-Drivers',
+	#package : 'Refactoring-UI',
+	#tag : 'Drivers'
+}
+
+{ #category : 'execution' }
+RBRenameVariableDriver >> applyChanges [
+	
+	refactoring privateTransform.
+	self flag: 'The next line is horrible, 
+					I must think of a better way to prevent changes from executing during tests'.
+	(previewPresenterClass isKindOf: StRefactoringPreviewPresenterMock class)
+		ifFalse: [ refactoring performChanges ]
```

Comment:
Maybe split this into two hooks, one for creating changes one for appying/performing them? Just a suggestion, hope it helps.

Reply:
I propose [this](https://github.com/pharo-project/pharo/pull/15774/commits/497c89f41a5fceeb671c20de825f150435ecce75) because I couldn't find a better way to split the changes creation and performing them, while being able to prevent the application of changes during tests and at the same time, the dependency on StRefactoringPreviewPresenterMock. 

I don't like much the idea of using a presenter only to apply changes and never opening it. 
What do you think?

Reply:
I agree with you, but I think this new way is much better and that we can integrate and iterate later..